tests/test_api.py

```diff
@@ -425,3 +425,11 @@ class CustomEnvironment(Environment):
         env = CustomEnvironment()
         tmpl = env.from_string("{{ foo }}")
         assert tmpl.render() == "resolve-foo"
+
+
+def test_overlay_enable_async(env):
+    assert not env.is_async
+    assert not env.overlay().is_async
+    env_async = env.overlay(enable_async=True)
+    assert env_async.is_async
```

Comment:
Missing test case of the concerning issue with `overlay` without any `enable_async` parameter , that is : 
```python
    assert env_async.overlay().is_async
 ```